// @aws-sdk/client-ecs/dist-cjs/ECS.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ECS = void 0;
const smithy_client_1 = require("@aws-sdk/smithy-client");
const CreateCapacityProviderCommand_1 = require("./commands/CreateCapacityProviderCommand");
const CreateClusterCommand_1 = require("./commands/CreateClusterCommand");
const CreateServiceCommand_1 = require("./commands/CreateServiceCommand");
const CreateTaskSetCommand_1 = require("./commands/CreateTaskSetCommand");
const DeleteAccountSettingCommand_1 = require("./commands/DeleteAccountSettingCommand");
const DeleteAttributesCommand_1 = require("./commands/DeleteAttributesCommand");
const DeleteCapacityProviderCommand_1 = require("./commands/DeleteCapacityProviderCommand");
const DeleteClusterCommand_1 = require("./commands/DeleteClusterCommand");
const DeleteServiceCommand_1 = require("./commands/DeleteServiceCommand");
const DeleteTaskDefinitionsCommand_1 = require("./commands/DeleteTaskDefinitionsCommand");
const DeleteTaskSetCommand_1 = require("./commands/DeleteTaskSetCommand");
const DeregisterContainerInstanceCommand_1 = require("./commands/DeregisterContainerInstanceCommand");
const DeregisterTaskDefinitionCommand_1 = require("./commands/DeregisterTaskDefinitionCommand");
const DescribeCapacityProvidersCommand_1 = require("./commands/DescribeCapacityProvidersCommand");
const DescribeClustersCommand_1 = require("./commands/DescribeClustersCommand");
const DescribeContainerInstancesCommand_1 = require("./commands/DescribeContainerInstancesCommand");
const DescribeServicesCommand_1 = require("./commands/DescribeServicesCommand");
const DescribeTaskDefinitionCommand_1 = require("./commands/DescribeTaskDefinitionCommand");
const DescribeTasksCommand_1 = require("./commands/DescribeTasksCommand");
const DescribeTaskSetsCommand_1 = require("./commands/DescribeTaskSetsCommand");
const DiscoverPollEndpointCommand_1 = require("./commands/DiscoverPollEndpointCommand");
const ExecuteCommandCommand_1 = require("./commands/ExecuteCommandCommand");
const GetTaskProtectionCommand_1 = require("./commands/GetTaskProtectionCommand");
const ListAccountSettingsCommand_1 = require("./commands/ListAccountSettingsCommand");
const ListAttributesCommand_1 = require("./commands/ListAttributesCommand");
const ListClustersCommand_1 = require("./commands/ListClustersCommand");
const ListContainerInstancesCommand_1 = require("./commands/ListContainerInstancesCommand");
const ListServicesByNamespaceCommand_1 = require("./commands/ListServicesByNamespaceCommand");
const ListServicesCommand_1 = require("./commands/ListServicesCommand");
const ListTagsForResourceCommand_1 = require("./commands/ListTagsForResourceCommand");
const ListTaskDefinitionFamiliesCommand_1 = require("./commands/ListTaskDefinitionFamiliesCommand");
const ListTaskDefinitionsCommand_1 = require("./commands/ListTaskDefinitionsCommand");
const ListTasksCommand_1 = require("./commands/ListTasksCommand");
const PutAccountSettingCommand_1 = require("./commands/PutAccountSettingCommand");
const PutAccountSettingDefaultCommand_1 = require("./commands/PutAccountSettingDefaultCommand");
const PutAttributesCommand_1 = require("./commands/PutAttributesCommand");
const PutClusterCapacityProvidersCommand_1 = require("./commands/PutClusterCapacityProvidersCommand");
const RegisterContainerInstanceCommand_1 = require("./commands/RegisterContainerInstanceCommand");
const RegisterTaskDefinitionCommand_1 = require("./commands/RegisterTaskDefinitionCommand");
const RunTaskCommand_1 = require("./commands/RunTaskCommand");
const StartTaskCommand_1 = require("./commands/StartTaskCommand");
const StopTaskCommand_1 = require("./commands/StopTaskCommand");
const SubmitAttachmentStateChangesCommand_1 = require("./commands/SubmitAttachmentStateChangesCommand");
const SubmitContainerStateChangeCommand_1 = require("./commands/SubmitContainerStateChangeCommand");
const SubmitTaskStateChangeCommand_1 = require("./commands/SubmitTaskStateChangeCommand");
const TagResourceCommand_1 = require("./commands/TagResourceCommand");
const UntagResourceCommand_1 = require("./commands/UntagResourceCommand");
const UpdateCapacityProviderCommand_1 = require("./commands/UpdateCapacityProviderCommand");
const UpdateClusterCommand_1 = require("./commands/UpdateClusterCommand");
const UpdateClusterSettingsCommand_1 = require("./commands/UpdateClusterSettingsCommand");
const UpdateContainerAgentCommand_1 = require("./commands/UpdateContainerAgentCommand");
const UpdateContainerInstancesStateCommand_1 = require("./commands/UpdateContainerInstancesStateCommand");
const UpdateServiceCommand_1 = require("./commands/UpdateServiceCommand");
const UpdateServicePrimaryTaskSetCommand_1 = require("./commands/UpdateServicePrimaryTaskSetCommand");
const UpdateTaskProtectionCommand_1 = require("./commands/UpdateTaskProtectionCommand");
const UpdateTaskSetCommand_1 = require("./commands/UpdateTaskSetCommand");
const ECSClient_1 = require("./ECSClient");
const commands = {
    CreateCapacityProviderCommand: CreateCapacityProviderCommand_1.CreateCapacityProviderCommand,
    CreateClusterCommand: CreateClusterCommand_1.CreateClusterCommand,
    CreateServiceCommand: CreateServiceCommand_1.CreateServiceCommand,
    CreateTaskSetCommand: CreateTaskSetCommand_1.CreateTaskSetCommand,
    DeleteAccountSettingCommand: DeleteAccountSettingCommand_1.DeleteAccountSettingCommand,
    DeleteAttributesCommand: DeleteAttributesCommand_1.DeleteAttributesCommand,
    DeleteCapacityProviderCommand: DeleteCapacityProviderCommand_1.DeleteCapacityProviderCommand,
    DeleteClusterCommand: DeleteClusterCommand_1.DeleteClusterCommand,
    DeleteServiceCommand: DeleteServiceCommand_1.DeleteServiceCommand,
    DeleteTaskDefinitionsCommand: DeleteTaskDefinitionsCommand_1.DeleteTaskDefinitionsCommand,
    DeleteTaskSetCommand: DeleteTaskSetCommand_1.DeleteTaskSetCommand,
    DeregisterContainerInstanceCommand: DeregisterContainerInstanceCommand_1.DeregisterContainerInstanceCommand,
    DeregisterTaskDefinitionCommand: DeregisterTaskDefinitionCommand_1.DeregisterTaskDefinitionCommand,
    DescribeCapacityProvidersCommand: DescribeCapacityProvidersCommand_1.DescribeCapacityProvidersCommand,
    DescribeClustersCommand: DescribeClustersCommand_1.DescribeClustersCommand,
    DescribeContainerInstancesCommand: DescribeContainerInstancesCommand_1.DescribeContainerInstancesCommand,
    DescribeServicesCommand: DescribeServicesCommand_1.DescribeServicesCommand,
    DescribeTaskDefinitionCommand: DescribeTaskDefinitionCommand_1.DescribeTaskDefinitionCommand,
    DescribeTasksCommand: DescribeTasksCommand_1.DescribeTasksCommand,
    DescribeTaskSetsCommand: DescribeTaskSetsCommand_1.DescribeTaskSetsCommand,
    DiscoverPollEndpointCommand: DiscoverPollEndpointCommand_1.DiscoverPollEndpointCommand,
    ExecuteCommandCommand: ExecuteCommandCommand_1.ExecuteCommandCommand,
    GetTaskProtectionCommand: GetTaskProtectionCommand_1.GetTaskProtectionCommand,
    ListAccountSettingsCommand: ListAccountSettingsCommand_1.ListAccountSettingsCommand,
    ListAttributesCommand: ListAttributesCommand_1.ListAttributesCommand,
    ListClustersCommand: ListClustersCommand_1.ListClustersCommand,
    ListContainerInstancesCommand: ListContainerInstancesCommand_1.ListContainerInstancesCommand,
    ListServicesCommand: ListServicesCommand_1.ListServicesCommand,
    ListServicesByNamespaceCommand: ListServicesByNamespaceCommand_1.ListServicesByNamespaceCommand,
    ListTagsForResourceCommand: ListTagsForResourceCommand_1.ListTagsForResourceCommand,
    ListTaskDefinitionFamiliesCommand: ListTaskDefinitionFamiliesCommand_1.ListTaskDefinitionFamiliesCommand,
    ListTaskDefinitionsCommand: ListTaskDefinitionsCommand_1.ListTaskDefinitionsCommand,
    ListTasksCommand: ListTasksCommand_1.ListTasksCommand,
    PutAccountSettingCommand: PutAccountSettingCommand_1.PutAccountSettingCommand,
    PutAccountSettingDefaultCommand: PutAccountSettingDefaultCommand_1.PutAccountSettingDefaultCommand,
    PutAttributesCommand: PutAttributesCommand_1.PutAttributesCommand,
    PutClusterCapacityProvidersCommand: PutClusterCapacityProvidersCommand_1.PutClusterCapacityProvidersCommand,
    RegisterContainerInstanceCommand: RegisterContainerInstanceCommand_1.RegisterContainerInstanceCommand,
    RegisterTaskDefinitionCommand: RegisterTaskDefinitionCommand_1.RegisterTaskDefinitionCommand,
    RunTaskCommand: RunTaskCommand_1.RunTaskCommand,
    StartTaskCommand: StartTaskCommand_1.StartTaskCommand,
    StopTaskCommand: StopTaskCommand_1.StopTaskCommand,
    SubmitAttachmentStateChangesCommand: SubmitAttachmentStateChangesCommand_1.SubmitAttachmentStateChangesCommand,
    SubmitContainerStateChangeCommand: SubmitContainerStateChangeCommand_1.SubmitContainerStateChangeCommand,
    SubmitTaskStateChangeCommand: SubmitTaskStateChangeCommand_1.SubmitTaskStateChangeCommand,
    TagResourceCommand: TagResourceCommand_1.TagResourceCommand,
    UntagResourceCommand: UntagResourceCommand_1.UntagResourceCommand,
    UpdateCapacityProviderCommand: UpdateCapacityProviderCommand_1.UpdateCapacityProviderCommand,
    UpdateClusterCommand: UpdateClusterCommand_1.UpdateClusterCommand,
    UpdateClusterSettingsCommand: UpdateClusterSettingsCommand_1.UpdateClusterSettingsCommand,
    UpdateContainerAgentCommand: UpdateContainerAgentCommand_1.UpdateContainerAgentCommand,
    UpdateContainerInstancesStateCommand: UpdateContainerInstancesStateCommand_1.UpdateContainerInstancesStateCommand,
    UpdateServiceCommand: UpdateServiceCommand_1.UpdateServiceCommand,
    UpdateServicePrimaryTaskSetCommand: UpdateServicePrimaryTaskSetCommand_1.UpdateServicePrimaryTaskSetCommand,
    UpdateTaskProtectionCommand: UpdateTaskProtectionCommand_1.UpdateTaskProtectionCommand,
    UpdateTaskSetCommand: UpdateTaskSetCommand_1.UpdateTaskSetCommand,
};
class ECS extends ECSClient_1.ECSClient {
}
exports.ECS = ECS;
(0, smithy_client_1.createAggregatedClient)(commands, ECS);
